build/build.go

```diff
@@ -520,6 +520,10 @@ func toSolveOpt(ctx context.Context, d driver.Driver, multiDriver bool, opt Opti
 		so.FrontendAttrs["platform"] = strings.Join(pp, ",")
 	}
 
+	if len(opt.Platforms) != 1 && d.Factory().Name() == "docker-container" {
```

Comment:
This does not look correct. Multi-platform builds with `docker-container` driver are definitely supported. If I understood your description correctly then the condition you want is to check if exporting is happening to docker. Also, we should never check for driver names. There is a features list for that but don't think you need that atm., just check for the exporter.